src/middleware/navigation/sign.out.base.url.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * Middleware to generate the sign-out baseUrl for each journey
+ */
+import { NextFunction, Request, Response } from "express";
+
+import * as config from "../../config";
+import { logger } from "../../utils/logger";
+import {
+  isRegistrationJourney,
+  getTransactionIdAndSubmissionIdFromOriginalUrl,
+  getUrlWithTransactionIdAndSubmissionId
+} from "../../utils/url";
+
+export const generateSignOutBaseUrl = (req: Request, res: Response, next: NextFunction) => {
+  logger.info("setting sign-out-base-url");
+  let signOutBaseUrl: string = config.UPDATE_AN_OVERSEAS_ENTITY_URL;
+  if (isRegistrationJourney(req)) {
+    const transactionIdAndSubmissionId = getTransactionIdAndSubmissionIdFromOriginalUrl(req);
```

Comment:
No check for this flag is required here as we use the existence of the key/value pairs for `transaction` and `submission` to determine what our signOutBaseUrl should be.

Reply:
But it looks like we're deciding what to do based on the presence of the ids. Normally these decisions are based on the feature flag check. Later, when we remove the feature flag (and search the code-base) we find all the occurrences and know what to change. That won't be the case here. I still think fundamentally that the decision should be based on the feature flag and if it's an Update/Remove journey with Redis removal enabled and no ids can be found then the code should error. At the moment, it won't error, it will set REGISTER_AN_OVERSEAS_ENTITY_URL as the URL.

Reply:
Had a chat and now convinced. Marking as resolved.